support-models/src/main/scala/com/gu/support/workers/ProductTypeRatePlans.scala

```diff
@@ -39,7 +39,8 @@ object ProductTypeRatePlans {
 
   implicit val digitalRatePlan: ProductTypeRatePlans[DigitalPack] = (product: DigitalPack, environment: TouchPointEnvironment, fixedTerm) =>
     product.catalogType.ratePlans.getOrElse(environment, Nil).find(productRatePlan =>
-      productRatePlan.billingPeriod == product.billingPeriod
+      (productRatePlan.billingPeriod == product.billingPeriod && productRatePlan.productOptions == product.productOptions) ||
+        (productRatePlan.productOptions == Corporate && product.productOptions == Corporate) // We don't care about the billing period for corporates
```

Comment:
I can see the logic here if I squint far enough, but I feel it would make more sense to branch based on what we are looking for.  I also think the naming could be improved.
rename `product` to `productToFind` and `productRatePlan` to `candidateProduct`
make it
```scala
if (productToFind.productOptions == Corporate) {
  candidateProduct.productOptions == Corporate
} else {
  (productRatePlan.billingPeriod == product.billingPeriod && productRatePlan.productOptions == product.productOptions)
}
```

Reply:
I'm going to leave this for now because the changes would be quite extensive - there are similar extension methods for the other product types as well.

Reply:
ok didn't realise it's a bigger pattern, makes sense to pick it up when the time is better 👍